src/main/scala/sigmastate/lang/Terms.scala

```diff
@@ -53,7 +54,10 @@ object Terms {
     val body: SValue
   }
 
-  case class ValNode(name: String, givenType: SType, body: SValue) extends Val {
+  case class ValNode(name: String,
+                     givenType: SType,
+                     body: SValue,
+                     override val sourceContext: Nullable[SourceContext] = Nullable.None) extends Val {
```

Comment:
@greenhat `sourceContext` cannot be parameter of case class constructor. It is not part of IR semantics, and shouldn't participate in `equals` and `hashCode`.

Reply:
Thanks, missed that. Fixed.